spec/controllers/articles_controller_spec.rb

```diff
@@ -0,0 +1,66 @@
+# frozen_string_literal: true
+
+require "rails_helper"
+require_relative "../support/devise"
+
+RSpec.describe ArticlesController, type: :controller do
+  user = Fabricate(:user, role: "user")
+  article = Fabricate(:article, user_id: user.id, status: "public")
+  params = { id: article.id }
+
+  def valid_article_attributes
+    { title: Faker::Lorem.sentence,
+      body: Faker::Lorem.paragraphs,
+      status: "public",
+      user_id: Fabricate(:user).id,
+      created_at: Time.zone.now }
+  end
+
+  shared_examples "response test" do |url, user_must_be_signed_in = false|
+    if user_must_be_signed_in
+      it "forbidden action for ##{url}" do
+        get url, params: params
+        expect(response.status).to eq(302)
+      end
+
+      it "redirects to root" do
+        get url, params: params
+        expect(response).to redirect_to(root_path)
+      end
+    else
+      it "returns successful responce for ##{url}" do
+        sign_in user
+        get url, params: params
+        expect(response.status).to eq(200)
+      end
+
+      it "redirects to template" do
+        sign_in user
+        get url, params: params
+        expect(response).to render_template(url)
+      end
+
```

Comment:
What else I could test except respond status and template render? Try to imitate user activity with forbidden actions? Should I use Capybara?

Reply:
> “For new Rails apps: we don’t recommend adding the rails-controller-testing gem to your application. The official recommendation of the Rails team and the RSpec core team is to write request specs instead.”

Reply:
You probably want to check the actual articles creation/edit.
Adding Capybara is good also to test in-browser behavior, let's have one capybara test.